### ISSUE TITLE
[Enhancement] Hide ScrollView scroll bars.

### ISSUE THREAD

**Issue by kingces95** [xamarin/Xamarin.Forms#1670]
## Rationale

Designers, in pursuit of their UX, may reasonably ask that the scroll bar be removed from a ScrollView.

see https://forums.xamarin.com/discussion/73353/how-to-hide-scroll-bar-in-scrollview
see https://stackoverflow.com/questions/9837003/how-to-hide-remove-the-scrollbar-in-uiscrollview-in-iphone

## Implementation

```csharp
public class ScrollView {
	public static readonly BindableProperty ShowVerticalScrollBar; // bool
	public static readonly BindableProperty ShowHorizontalScrollBar; // bool
}
```

## Expected Result

Hides vertical and/or horizontal scroll bar.

## Implications for CSS

None.

## Backward Compatibility

None. By default, properties are true.

## Difficulty : Easy

**Comment by jassmith** [comment]
This should be implemented with an enum with some kind of ScrollBarVisibility.

```csharp
public enum ScrollBarVisibility
{
  Default,
  Always,
  Never,
}
```

seems like the logical choice forward. There will likely be other visibility modes cropping up and a need to make sure there is a default in case default behaviors differ on different platforms.

**Comment by bmacombe** [comment]
@davidortinau I'll pick this one up since the repeater control is in limbo at the moment.

**Comment by jassmith** [comment]
Hey checking in to see if this is still being worked on. If not we are going to move it back to being up for grabs.

**Comment by bmacombe** [comment]
Its done...PR is in the community fork.  Let me know what I need to do

**Comment by bmacombe** [comment]
https://github.com/XamarinFormsCommunity/Xamarin.Forms/pull/6

**Comment by jassmith** [comment]
Can you see why we are not getting the PR upstreamed please? The community fork cant be merged wholesale so we need to receive your PR upstream somehow.

**Comment by jassmith** [comment]
Thank you for the link, I will add a review there :)

**Comment by bmacombe** [comment]
Thanks!  I'll take a look...my first time contributing like this, so I'm a bit lost :)

**Comment by bmacombe** [comment]
I'll clean up the code in the community fork based on the review and then create a PR here.

**Comment by bmacombe** [comment]
https://github.com/xamarin/Xamarin.Forms/pull/1910

PR with the requested changes from jassmith's review.

**Comment by NickDevoctomy** [comment]
I've just upgraded my Xamarin.Forms package to 3.1.0.583944 and I now see this property available in the ScrollViewer.  It does nothing though, is that expected?



```
<ScrollView Orientation="Horizontal" HorizontalScrollBarVisibility="Never" VerticalScrollBarVisibility="Never">
<StackLayout Orientation="Horizontal" Spacing="0">
<local:TabButton WidthRequest="100" ButtonGroup="Test"></local:TabButton>
<local:TabButton WidthRequest="100" ButtonGroup="Test"></local:TabButton>
<local:TabButton WidthRequest="100" ButtonGroup="Test"></local:TabButton>
<local:TabButton WidthRequest="100" ButtonGroup="Test"></local:TabButton>
<local:TabButton WidthRequest="100" ButtonGroup="Test"></local:TabButton>
<local:TabButton WidthRequest="100" ButtonGroup="Test"></local:TabButton>
<local:TabButton WidthRequest="100" ButtonGroup="Test"></local:TabButton>
</StackLayout>
</ScrollView>
```

I still see the scroll bar exactly as I did in 3.0.  According to Visual Studio this is a stable build but it's showing a bunch of other "errors" with the xaml above with this version.  Although I can get around this by closing the xaml file, it compiles fine.  I'll report that as a separate issue.

**Comment by bmacombe** [comment]
@NickDevoctomy I'm curious what platforms you are seeing this on?  Wondering if it's Android or others also?

**Comment by NickDevoctomy** [comment]
Just UWP it seems, I can't test iOS, but just tested Android and it looks great.

**Comment by bmacombe** [comment]
Interesting...I was sure it would be android if any had issues.  I'll try to take a look see if I can find what's up.

**Comment by bmacombe** [comment]
@NickDevoctomy I just pulled the current master branch of XF and looked at the control gallery app on UWP and the scroll bar visibility is being updated as expected.  Could you post a repo?

**Comment by NickDevoctomy** [comment]
I have an isolated application that has the issue in I can post...

**Comment by NickDevoctomy** [comment]
https://1drv.ms/u/s!AlN90gKdry3-tKgDRaCPE7-6FAzBvg

Just run it and then resize the window so that the tabs don't fit on the page.  Looks fine when I run it on my S8, but on UWP the scroll bars appear.

**Comment by bmacombe** [comment]
@NickDevoctomy FYI, I know what is happening...but not why :)

While I can step though the ScrollViewRenderer for UWP and see that during the creation of the native control the visibility is being set correctly, it's not being respected or is reset at some point before display.  This scenario isn't tested in the Control Gallery because all of that stuff is done from code behind and no xaml.

I added a button to your TabView that toggles the visibility after it loads and it works as expected.

I also tried setting the visibility property in the constructor of the TabView with the same result as it being set in Xaml, so it's a control init problem for sure.

I'll keep digging a bit.
